### PR TITLE
Remove star imports

### LINKED ISSUE #1057: Remove star imports
# About

image
We currently have a lot of star imports in the project.
They should be replaced by normal imports and maybe add a check to sonar to prevent this from happening.

### PR DESCRIPTION
**Pull request by SimonVerhoeven**
Removes star imports. Resolves #1057

### CODE REVIEW

**Review by tj-wazei — APPROVED** [review]
LGTM

I'm not sure if merging this will resolve the issue but we can re-open it I guess, just to update Sonar/any linting we have to flag * imports.

**Review by Taz03 — DISMISSED** [review]
add a sonar / checkstyle check to prevent this from happening again (as stated in the issue)

**Comment by SimonVerhoeven** [comment]
> LGTM
> 
> I'm not sure if merging this will close the issue but we can re-open it I guess, just to update Sonar/any linting we have to flag * imports.

True, it's a partial resolution. But generally I prefer configuration and code changes in 2 separate PRs. As this will have no direct impact, but merging it will prevent PRs. While changing sonar configuration might for open PRs.

File: application/src/main/java/org/togetherjava/tjbot/features/moderation/audit/ModAuditLogRoutine.java [hunk 1]
```diff
@@ -300,7 +310,7 @@ private boolean containsMutedRole(AuditLogEntry entry, AuditLogKey key) {
             .orElse(List.of());
         return roleChanges.stream()
             .map(Map::entrySet)
-            .flatMap(Collection::stream)
+            .flatMap(java.util.Collection::stream)
```
Comment (SquidXTV) [inline application/src/main/java/org/togetherjava/tjbot/features/moderation/audit/ModAuditLogRoutine.java:313]: Fix this

File: application/src/test/java/org/togetherjava/tjbot/features/mathcommands/TeXCommandTest.java [hunk 2]
```diff
@@ -14,7 +14,9 @@
 import java.util.ArrayList;
 import java.util.List;
 
-import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.ArgumentMatchers.contains;
+import static org.mockito.ArgumentMatchers.startsWith;
```
Comment (SquidXTV) [inline application/src/test/java/org/togetherjava/tjbot/features/mathcommands/TeXCommandTest.java:19]: Not sure about mockito imports tbh, I would understand if we keep star imports here. Though that is not really consistent. Same with assertions in other files.

**Review by SquidXTV — CHANGES_REQUESTED** [review]

File: application/src/test/java/org/togetherjava/tjbot/features/mathcommands/TeXCommandTest.java (hunk 2 above)
Comment (SimonVerhoeven) [reply application/src/test/java/org/togetherjava/tjbot/features/mathcommands/TeXCommandTest.java:19]: I'd do it like this for consistency's sake.

**Review by tj-wazei — APPROVED** [review]

**Pull request merged**